src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorProjectChangePublisher.cs

```diff
@@ -206,9 +227,27 @@ protected virtual void SerializeToFile(ProjectSnapshot projectSnapshot, string p
             tempFileInfo.MoveTo(publishFilePath);
         }
 
-        private async Task PublishAfterDelayAsync(string projectFilePath)
+        protected virtual bool ShouldSerialize(string configurationFilePath)
+        {
+            // I split this up as a premature optimization to avoid making unneeded queries.
+            if (!File.Exists(configurationFilePath))
+            {
+                return true;
+            }
+
+            var status = _operationProgressStatusService?.GetStageStatusForSolutionLoad(CommonOperationProgressStageIds.Intellisense);
+
+            if (status is null)
+            {
+                return true;
```

Comment:
What does `null` represent? I assume done since we're returning `true` here? Or does it only happen if we couldn't resolve our operationProgressStatusService?

Reply:
As far as I know it can only be null when we couldn't resolve `operationProgressStatusService`.  I thought it better to default to serializing, though as far as I know this service should always be available.